pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -18,64 +19,92 @@ assert tclSupport -> tcl != null;
-    ++ optional (!luaSupport)   "-DENABLE_LUA=OFF"
-    ++ optional (!perlSupport)  "-DENABLE_PERL=OFF"
-    ++ optional (!rubySupport)  "-DENABLE_RUBY=OFF"
-    ++ optional (!tclSupport)   "-DENABLE_TCL=OFF"
-    ;
+      buildInputs = with stdenv.lib; [
+          ncurses python openssl aspell gnutls zlib curl pkgconfig
+          libgcrypt pycrypto makeWrapper
+          cmake
+          asciidoctor
+          ]
+        ++ optionals stdenv.isDarwin [ pync libobjc libresolv ]
+        ++ optional  guileSupport    guile
+        ++ optional  luaSupport      lua5
+        ++ optional  perlSupport     perl
+        ++ optional  rubySupport     ruby
+        ++ optional  tclSupport      tcl
+        ++ extraBuildInputs;
 
-  buildInputs = with stdenv.lib; [
-      ncurses python openssl aspell gnutls zlib curl pkgconfig
-      libgcrypt pycrypto makeWrapper
-      cmake
-      asciidoctor
-      ]
-    ++ optionals stdenv.isDarwin [ pync libobjc libresolv ]
-    ++ optional  guileSupport    guile
-    ++ optional  luaSupport      lua5
-    ++ optional  perlSupport     perl
-    ++ optional  rubySupport     ruby
-    ++ optional  tclSupport      tcl
-    ++ extraBuildInputs;
+      NIX_CFLAGS_COMPILE = "-I${python}/include/${python.libPrefix}"
+        # Fix '_res_9_init: undefined symbol' error
+        + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1 -lresolv");
 
-  NIX_CFLAGS_COMPILE = "-I${python}/include/${python.libPrefix}"
-    # Fix '_res_9_init: undefined symbol' error
-    + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1 -lresolv");
+      postInstall = with stdenv.lib; ''
+        NIX_PYTHONPATH="$out/lib/${python.libPrefix}/site-packages"
+        wrapProgram "$out/bin/weechat" \
+          ${optionalString perlSupport "--prefix PATH : ${perl}/bin"} \
+          --prefix PATH : ${pythonPackages.python}/bin \
+          --prefix PYTHONPATH : "$PYTHONPATH" \
+          --prefix PYTHONPATH : "$NIX_PYTHONPATH"
+        for o in $outputs; do
+          from=$out/lib/weechat/plugins/$o.so
+          to=''${!o}/lib/weechat/plugins/$o.so
+          if [[ -e $from ]]
+          then
+            mkdir -p $(dirname $to)
+            mv $from $to
+          fi
+        done
+      '';
 
-  postInstall = with stdenv.lib; ''
-    NIX_PYTHONPATH="$out/lib/${python.libPrefix}/site-packages"
-    wrapProgram "$out/bin/weechat" \
-      ${optionalString perlSupport "--prefix PATH : ${perl}/bin"} \
-      --prefix PATH : ${pythonPackages.python}/bin \
-      --prefix PYTHONPATH : "$PYTHONPATH" \
-      --prefix PYTHONPATH : "$NIX_PYTHONPATH"
+      meta = {
+        homepage = http://www.weechat.org/;
+        description = "A fast, light and extensible chat client";
+        license = stdenv.lib.licenses.gpl3;
+        maintainers = with stdenv.lib.maintainers; [ lovek323 garbas the-kenny ];
+        platforms = stdenv.lib.platforms.unix;
+      };
+    };
+in weechat // {
```

Comment:
Right, that makes sense... so something like d421f4aefcaddf0ec07582ffb62fa6f625dd8f96 ?

(usage: `weechat.override {withPlugins = ["perl" "python" "ruby"];}`)